packages/core/saved-objects/core-saved-objects-base-server-internal/src/model_version/get_version_delta.ts

```diff
@@ -81,18 +87,9 @@ const getTypeDelta = ({
   currentVersions: VirtualVersionMap;
   targetVersions: VirtualVersionMap;
 }): ModelVersionDeltaTypeResult => {
-  const currentVersion = currentVersions[type];
-  const targetVersion = targetVersions[type];
-  if (currentVersion === undefined || targetVersion === undefined) {
-    // should never occur given we've been checking consistency numerous times before getting there
-    // but better safe than sorry.
-    throw new Error(
-      `Consistency error: trying to generate delta with missing entry for type ${type}`
-    );
-  }
```

Comment:
"should never occur", right. I wonder who wrote that code :trollface: .

All other parts of the algo properly support type additions/removal when doing version comparisons, it was just overlooked that we needed to do the same for version deltas (and that having a version not present in the current or target map is a normal scenario)